common4j/src/main/com/microsoft/identity/common/java/cache/BrokerOAuth2TokenCache.java

```diff
@@ -582,6 +585,7 @@ public ICacheRecord load(@NonNull final String clientId,
         } else {
             resultRecord = targetCache.load(
                     clientId,
+                    applicationIdentifier,
```

Comment:
where are we saving this value?

Reply:
> @shoatman would you mind describing in detail
> 
> 1. the reason/scenario that drives this change?
> 2. How to navigate through the flow affected by the change (IMHO, the cache layer is pretty convoluted).
> 
> I think it'll be helpful for reviewers, and as a future reference.

Happy to provide additional context.  I think I added some design documentation on the cache to common already as a separate PR that Brian reviewed and approved.  I'll add a link to that.

Reply:
Dug into the code... this is very not obvious... the modification to the "save" happens in generateCacheKey().